### PR TITLE
Issue #2662 "Unnecessary boxing conversions"

### PR DESCRIPTION
**Pull request by mperktold**
Removed some unnecessary conversions from primitive to boxed types.

For finding them in Eclipse, I activated warnings for "Boxing and unboxing conversions", filtered for those in the "Problems" view, and quickly went through them.

### CODE REVIEW

File: jetty-infinispan/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanSessionDataStore.java [hunk 1]
```diff
@@ -258,7 +259,7 @@ public boolean exists(String id) throws Exception
     {
         // TODO find a better way to do this that does not pull into memory the
         // whole session object
-        final AtomicReference<Boolean> reference = new AtomicReference<>();
+        final AtomicBoolean reference = new AtomicBoolean();
```
Comment (joakime) [inline jetty-infinispan/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanSessionDataStore.java]: This seems out of scope of the PR.

Comment (gregw) [reply jetty-infinispan/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanSessionDataStore.java]: Not really, as it changes the method signatures where this is used, so they can avoid boxing

**Comment by gregw** [comment]
#2662
